Assets/MixedRealityToolkit/Utilities/Editor/DependencyWindow.cs

```diff
@@ -0,0 +1,339 @@
+
+        private const string windowTitle = "Dependency Window";
+        private const string guidPrefix = "guid: ";
+        private const string nullGuid = "00000000000000000000000000000000";
+        private const int guidCharacterCount = 32;
+
+        private readonly string[] assetsWithDependencies =
+        {
+            ".unity",
+            ".prefab",
+            ".asset",
+            ".mat",
+            ".anim",
+            ".controller"
+        };
+
+        [MenuItem("Mixed Reality Toolkit/Utilities/Dependency Window", false, 3)]
+        private static void ShowWindow()
+        {
+            var window = GetWindow<DependencyWindow>();
+            window.titleContent = new GUIContent(windowTitle, EditorGUIUtility.IconContent("d_EditCollider").image);
+            window.minSize = new Vector2(520.0f, 380.0f);
+            window.RefreshAssetGraph();
+            window.Show();
+        }
+
+        private void OnGUI()
+        {
+            MixedRealityEditorUtility.RenderMixedRealityToolkitLogo();
+
+            EditorGUILayout.LabelField("Mixed Reality Toolkit Dependency Window", new GUIStyle() {fontSize = 12, fontStyle = FontStyle.Bold });
+            EditorGUILayout.LabelField("This tool displays how assets reference and depend on each other. Dependencies are calculated by parsing guids within project YAML files, code dependencies are not considered.", EditorStyles.wordWrappedLabel);
+
+            EditorGUILayout.Space();
+
+            if (EditorSettings.serializationMode != SerializationMode.ForceText)
+            {
+                EditorGUILayout.HelpBox("Dependencies can only be tracked with text assets. Please change the project serialization mode to \"Force Text\" via Edit->Project Settings->Editor->Asset Serialization", MessageType.Error);
+            }
+
+            assetSelection = EditorGUILayout.ObjectField("Asset Selection", assetSelection, typeof(Object), false);
+            maxDisplayDepth = EditorGUILayout.IntSlider("Max Display Depth", maxDisplayDepth, 1, 32);
+
+            EditorGUILayout.BeginHorizontal();
+            {
+                if (dependencyGraph.Count == 0)
+                {
+                    GUILayout.Label(string.Format("The dependency graph contains 0 assets, please refresh the dependency graph.", dependencyGraph.Count));
+                }
+                else
+                {
+                    GUILayout.Label(string.Format("The dependency graph contains {0:n0} assets and took {1:0.00} seconds to build.", dependencyGraph.Count, assetGraphRefreshTime));
+                }
+
+                if (GUILayout.Button("Refresh"))
+                {
+                    RefreshAssetGraph();
+                }
+            }
+            EditorGUILayout.EndHorizontal();
+
+            EditorGUILayout.Space();
+
+            GUILayout.BeginVertical("Box");
+            {
+                GUILayout.Label("Dependency Graph", EditorStyles.boldLabel);
```

Comment:
Can you to using EditorGUILayout instead of GUILayout? 1) to be consistent with other inspector/editor files and 2) EditoGUILayout generally has more features (with the very odd weird exception of GUILayout.Button)

https://docs.unity3d.com/ScriptReference/EditorGUILayout.LabelField.html

Reply:
I switched over as many as I could (sans Button, parametrized Space, and a few Labels that require special alignment). That's so weird their isn't a EditorGUILayout Button?